core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/util/JsonUtil.java

```diff
@@ -39,20 +38,23 @@ public class JsonUtil {
 
     private JsonUtil() {}
 
-    public static JSONArray objectsToJsonArray(Collection<?> objects) {
-        JSONArray jsonArray = new JSONArray();
+    public static JsonArray objectsToJsonArray(Collection<?> objects) {
+        JsonArrayBuilder jsonArrayBuilder = Json.createArrayBuilder();
         objects.forEach(object -> {
             try {
                 String json = OBJECT_MAPPER.writeValueAsString(object);
-                jsonArray.put(new JSONObject(json));
-            } catch (JsonProcessingException | JSONException e) {
+                try (JsonReader jsonReader = Json.createReader(new StringReader(json))) {
+                    JsonObject jsonObject = jsonReader.readObject();
+                    jsonArrayBuilder.add(jsonObject);
+                }
+            } catch (JsonProcessingException e) {
                 LOG.error("Failed to convert gridResources to JSON", e);
             }
         });
-        return jsonArray;
+        return jsonArrayBuilder.build();
     }
 
-    public static <T> T jsonToModel(JSONObject json, Class<T> modelClass) {
+    public static <T> T jsonToModel(JsonObject json, Class<T> modelClass) {
```

Comment:
To replace this whole method:

```suggestion
    public static <T> T jsonToModel(JsonNode json, Class<T> modelClass) {
        try (InputStream is = new ByteArrayInputStream(json.toString().getBytes(StandardCharsets.UTF_8))) {
            final JavaType type = OBJECT_MAPPER.getTypeFactory().constructType(modelClass);
            return OBJECT_MAPPER.readValue(is, type);
        } catch (IOException e) {
            LOG.error("Failed to map json to model", e);
        }
        return null;
    }
```

Reply:
done